be/src/io/fs/s3_file_writer.cpp

```diff
@@ -17,221 +17,232 @@
 
 #include "io/fs/s3_file_writer.h"
 
-#include <aws/core/Aws.h>
 #include <aws/core/utils/HashingUtils.h>
+#include <aws/core/utils/memory/stl/AWSStreamFwd.h>
+#include <aws/core/utils/memory/stl/AWSStringStream.h>
 #include <aws/s3/S3Client.h>
 #include <aws/s3/model/AbortMultipartUploadRequest.h>
 #include <aws/s3/model/CompleteMultipartUploadRequest.h>
+#include <aws/s3/model/CompletedPart.h>
 #include <aws/s3/model/CreateMultipartUploadRequest.h>
-#include <aws/s3/model/DeleteObjectRequest.h>
-#include <aws/s3/model/DeleteObjectsRequest.h>
-#include <aws/s3/model/GetObjectRequest.h>
 #include <aws/s3/model/UploadPartRequest.h>
-#include <fmt/core.h>
-#include <sys/uio.h>
 
-#include <cerrno>
+#include <atomic>
+#include <memory>
+#include <utility>
 
-#include "common/compiler_util.h"
+#include "common/logging.h"
 #include "common/status.h"
-#include "gutil/macros.h"
+#include "io/fs/file_system.h"
 #include "io/fs/file_writer.h"
-#include "io/fs/path.h"
 #include "io/fs/s3_file_system.h"
-#include "util/doris_metrics.h"
-
-using Aws::S3::Model::AbortMultipartUploadRequest;
-using Aws::S3::Model::CompletedPart;
-using Aws::S3::Model::CompletedMultipartUpload;
-using Aws::S3::Model::CompleteMultipartUploadRequest;
-using Aws::S3::Model::CreateMultipartUploadRequest;
-using Aws::S3::Model::DeleteObjectRequest;
-using Aws::S3::Model::UploadPartRequest;
-using Aws::S3::Model::UploadPartOutcome;
+#include "io/fs/s3_file_write_bufferpool.h"
+#include "util/runtime_profile.h"
+#include "util/s3_util.h"
 
 namespace doris {
 namespace io {
+using namespace Aws::S3::Model;
+using Aws::S3::S3Client;
 
-// max size of each part when uploading: 5MB
-static const int MAX_SIZE_EACH_PART = 5 * 1024 * 1024;
-static const char* STREAM_TAG = "S3FileWriter";
-
-S3FileWriter::S3FileWriter(Path path, std::shared_ptr<Aws::S3::S3Client> client,
+S3FileWriter::S3FileWriter(Path&& path, std::string&& key, std::shared_ptr<S3Client> client,
                            const S3Conf& s3_conf, FileSystemSPtr fs)
-        : FileWriter(std::move(path), fs), _client(client), _s3_conf(s3_conf) {
-    DorisMetrics::instance()->s3_file_open_writing->increment(1);
-    DorisMetrics::instance()->s3_file_writer_total->increment(1);
-}
+        : FileWriter(std::move(path), std::move(fs)),
+          _bucket(s3_conf.bucket),
+          _key(std::move(key)),
+          _client(std::move(client)) {}
 
 S3FileWriter::~S3FileWriter() {
-    if (_opened) {
-        close();
+    if (!_closed) {
+        abort();
```

Comment:
You change the `close()` to `abort()` there. Does it mean that we must call `close()` explicitly, otherwise the write operation may be aborted?
If yes, better to add comment for S3FileWriter